core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -222,6 +222,7 @@ private[spark] object ApiRootResource {
 private[spark] trait UIRoot {
   def getSparkUI(appKey: String): Option[SparkUI]
   def getApplicationInfoList: Iterator[ApplicationInfo]
+  def getApplicationInfoList(limit: Int): Iterator[ApplicationInfo]
```

Comment:
The default value doesn't help because 
1) The parameter **limit** that you have added in /applications endpoint is an **Integer**. So it can be null. When it is null, the default value won't help. Still need to differentiate null with not null.
2) When limit is a negative number, take(-1) is same as take(0) which returns an empty list. So I still need to judge whether it is less than or equals to 0.